TeachingRecordSystem/tests/TeachingRecordSystem.Api.Tests/V3/VNext/SetQtlsDateRequestTests.cs

```diff
@@ -0,0 +1,1133 @@
+
+        // Assert
+        Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
+    }
+
+    //2
+    [Fact]
+    public async Task Put_ValidQTSDateWithNoExistingQTSDate_ReturnsExpectedResult()
+    {
+        // Arrange
+        var qtlsDate = new DateOnly(2020, 01, 01);
+        var requestId = Guid.NewGuid().ToString();
+        var firstName = Faker.Name.First();
+        var middleName = Faker.Name.Middle();
+        var lastName = Faker.Name.Last();
+        var dateOfBirth = new DateOnly(1990, 01, 01);
+        var email = Faker.Internet.Email();
+        var nationalInsuranceNumber = Faker.Identification.UkNationalInsuranceNumber();
+
+        var existingContact = await TestData.CreatePerson(p => p
+            .WithTrn(hasTrn: true)
+            .WithFirstName(firstName)
+            .WithMiddleName(middleName)
+            .WithLastName(lastName)
+            .WithDateOfBirth(dateOfBirth)
+            .WithEmail(email)
+            .WithNationalInsuranceNumber(nationalInsuranceNumber: nationalInsuranceNumber));
+
+        var requestBody = CreateJsonContent(new { QTSDate = qtlsDate });
+        var request = new HttpRequestMessage(HttpMethod.Put, $"v3/persons/{existingContact.Trn}/qtls")
+        {
+            Content = requestBody
+        };
+
+        // Act
+        var response = await GetHttpClientWithApiKey().SendAsync(request);
+        var contact = XrmFakedContext.CreateQuery<Contact>().FirstOrDefault(x => x.ContactId == existingContact.PersonId);
+        var task = XrmFakedContext.CreateQuery<Core.Dqt.Models.Task>().FirstOrDefault(x => x.RegardingObjectId.Id == existingContact.PersonId && x.Subject == "Notification for SET QTLS data collections team");
+
+        // Assert
+        await AssertEx.JsonResponseEquals(
+            response,
+            expected: new
+            {
+                existingContact.Trn,
+                QTSDate = qtlsDate
+
+            },
+            expectedStatusCode: 200);
+
+        Assert.Null(task);
+        Assert.Equal(dfeta_InductionStatus.Exempt, contact!.dfeta_InductionStatus);
+        Assert.Equal(qtlsDate, contact!.dfeta_qtlsdate.ToDateOnlyWithDqtBstFix(isLocalTime: false));
+    }
+
+    //3
+    [Fact]
+    public async Task Put_ClearExistingQTLSDateWithNoQTS_ReturnsExpectedResult()
+    {
+        // Arrange
+        var qtlsDate = default(DateOnly?);
+        var existingContact = await TestData.CreatePerson(p => p
+            .WithTrn(hasTrn: true)
+            .WithQtlsDate(new DateOnly(2020, 01, 01)));
+
+        var requestBody = CreateJsonContent(new { QTSDate = qtlsDate });
+        var request = new HttpRequestMessage(HttpMethod.Put, $"v3/persons/{existingContact.Trn}/qtls")
+        {
+            Content = requestBody
+        };
+
+        // Act
+        var response = await GetHttpClientWithApiKey().SendAsync(request);
+        var contact = XrmFakedContext.CreateQuery<Contact>().FirstOrDefault(x => x.ContactId == existingContact.PersonId);
```

Comment:
```suggestion
        var contact = XrmFakedContext.CreateQuery<Contact>().Single(x => x.ContactId == existingContact.PersonId);
```
Can we also move this into the `Assert` section like the test above?